atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -288,8 +285,32 @@ public void compactInternally() {
                 throw new RuntimeException(e);
             } finally {
                 log.info("Call to KVS.compactInternally on table {} took {} ms.",
-                        tableRef, timer.elapsed(TimeUnit.MILLISECONDS));
+                        LoggingArgs.tableRef(tableRef), timer.elapsed(TimeUnit.MILLISECONDS));
             }
         }
     }
+
+    private void runShrinkSpaceCompact() throws TableMappingNotFoundException {
+        Stopwatch shrinkAndCompactTimer = Stopwatch.createStarted();
+        conns.get().executeUnregisteredQuery(
+                "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
+                        + " SHRINK SPACE COMPACT");
+        log.info("Call to SHRINK SPACE COMPACT on table {} took {} ms.",
+                tableRef, shrinkAndCompactTimer.elapsed(TimeUnit.MILLISECONDS));
+    }
+
+    private void runShrinkSpace() throws TableMappingNotFoundException {
+        log.warn("Starting to compact table {}. "
+                        + "This might lock this table for reads and writes a long period of time.",
+                LoggingArgs.tableRef(tableRef));
+
+        Stopwatch shrinkTimer = Stopwatch.createStarted();
+        conns.get().executeUnregisteredQuery(
+                "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
+                        + " SHRINK SPACE");
+
+        log.info("Call to SHRINK SPACE on table {} took {} ms."
+                        + " This implies that locks on the entire table were held for this period.",
+                tableRef, shrinkTimer.elapsed(TimeUnit.MILLISECONDS));
```

Comment:
OK. Feels like the slow-logging kind of belongs in another layer, but certainly outside of the scope of this